addons/delivery/models/stock_picking.py

```diff
@@ -228,7 +228,8 @@ def _set_delivery_package_type(self):
     def send_to_shipper(self):
         self.ensure_one()
         res = self.carrier_id.send_shipping(self)[0]
-        if self.carrier_id.free_over and self.sale_id and self.sale_id._compute_amount_total_without_delivery() >= self.carrier_id.amount:
+        amount_without_delivery = self.sale_id._compute_amount_total_without_delivery()
+        if self.carrier_id.free_over and self.sale_id and self.carrier_id._compute_currency(self.sale_id, amount_without_delivery, 'pricelist_to_company') >= self.carrier_id.amount:
```

Comment:
```suggestion
        if self.carrier_id.free_over and self.sale_id:
            amount_without_delivery = self.sale_id._compute_amount_total_without_delivery()
            if self.carrier_id._compute_currency(
                    self.sale_id, amount_without_delivery, 'pricelist_to_company'
            ) >= self.carrier_id.amount:
```
Here we'll have ValueError: Expected singleton: sale.order() if picking has no sale_id
See https://github.com/odoo/odoo/pull/111948 for details

Reply:
Hi, thank you for the update, please see: https://github.com/odoo/odoo/pull/112069